packages/patterns/integration/ct-checkbox-simple.test.ts

```diff
@@ -0,0 +1,142 @@
+import { env } from "@commontools/integration";
+import { sleep } from "@commontools/utils/sleep";
+import { ShellIntegration } from "@commontools/integration/shell-utils";
+import { afterAll, beforeAll, describe, it } from "@std/testing/bdd";
+import { join } from "@std/path";
+import { assert, assertEquals } from "@std/assert";
+import { Identity } from "@commontools/identity";
+import { CharmsController } from "@commontools/charm/ops";
+
+const { API_URL, FRONTEND_URL, SPACE_NAME } = env;
+
+describe("ct-checkbox-simple integration test", () => {
+  const shell = new ShellIntegration();
+  shell.bindLifecycle();
+
+  let charmId: string;
+  let identity: Identity;
+  let cc: CharmsController;
+
+  beforeAll(async () => {
+    identity = await Identity.generate({ implementation: "noble" });
+    cc = await CharmsController.initialize({
+      spaceName: SPACE_NAME,
+      apiUrl: new URL(API_URL),
+      identity: identity,
+    });
+    const charm = await cc.create(
+      await Deno.readTextFile(
+        join(
+          import.meta.dirname!,
+          "..",
+          "ct-checkbox-simple.tsx",
+        ),
+      ),
+    );
+    charmId = charm.id;
+  });
+
+  afterAll(async () => {
+    if (cc) await cc.dispose();
+  });
+
+  it("should load the ct-checkbox-simple charm", async () => {
+    const page = shell.page();
+    await shell.goto({
+      frontendUrl: FRONTEND_URL,
+      spaceName: SPACE_NAME,
+      charmId,
+      identity,
+    });
+    await page.waitForSelector("ct-checkbox", { strategy: "pierce" });
+  });
+
+  it("should show disabled content initially", async () => {
+    const page = shell.page();
+
+    // Wait for component to stabilize
+    await sleep(1000);
+
+    // Initially, feature should be disabled
+    const disabledContent = await page.$("[data-testid='disabled-content']", {
```

Comment:
Using `page.waitForSelector(..)` here has a built in wait loop -- unless `[data-testid='disabled-content']` is already in the DOM, can avoid the explicit sleep above and use `waitForSelector` (which either returns the element, or throws, so we don't need assertions to guard against `undefined`) -- I think this is the only potential racey spot AFAICT